apps/os/backend/agent/mcp/mcp-event-hooks.ts

```diff
@@ -382,6 +396,39 @@ export async function handleMCPConnectRequest(
 
   mcpConnectionCache.managers.set(cacheKey, manager);
 
+  // Ensure mcpConnection row exists for this connection
+  // This upserts based on the partial unique indexes for personal/company modes
+  await db.transaction(async (tx) => {
+    // Check if connection already exists
+    const existingConn = await tx.query.mcpConnection.findFirst({
+      where: and(
+        eq(schema.mcpConnection.estateId, estateId),
+        eq(schema.mcpConnection.serverUrl, serverUrl),
+        mode === "personal"
+          ? and(eq(schema.mcpConnection.mode, "personal"), eq(schema.mcpConnection.userId, userId))
+          : eq(schema.mcpConnection.mode, "company"),
+      ),
```

Comment:
I think we can simplify this logic?
```suggestion
        eq(schema.mcpConnection.mode, mode),
        eq(schema.mcpConnection.userId, userId || null),
```